### DISCUSSION TITLE
Updating DIND scenario

### DISCUSSION THREAD

**Discussion by nmiekley** [madhuakula/kubernetes-goat#127] [General]
I would like to talk about updating the [DIND scenario](https://madhuakula.com/kubernetes-goat/docs/scenarios/scenario-2/docker-in-docker-exploitation-in-kubernetes-containers). 

The original idea is that docker is running on the worker nodes and managing the containers for kubernetes. However, with [kubernetes deprecation for docker shim](https://kubernetes.io/blog/2020/12/02/dockershim-faq/) this is no longer the case. Now, either containerd or cri-o is used as container runtime for kubernetes. 

The same exploit can still be done, we only have to use the containerd or cri-o sockets.

* Should we update the scenario? I already did first steps in https://github.com/nmiekley/kubernetes-goat
* Is this scenario still relevant? And is there a name for it?

Btw, the current setup also has a funny effect in KIND: here the docker daemon you gain control over is the one on your machine running the kind node. You can still get control of the KIND node and the containerd inside, but that is not really a production relevant scenario.

**Comment by nmiekley** [comment]
I didn't see https://github.com/madhuakula/kubernetes-goat/issues/95 earlier. I think that would also be solved by this. I only tested it in KIND and AWS yet.

**Comment by madhuakula** [comment]
Thank you for the discussion, fix and updates. Appreciate it @nmiekley 🙏
